### ISSUE TITLE
[OpenBMC] rpower should detect when the BMC is in a service state 

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#3991]
I'm not sure whether this will happen in production, but we should detect when the bmc is in a "service" state, for example when a CRONUS box is connected to it for debug. 

It would be good to do this anyway so that in our development, we know when a machine is in service mode or whether the compute side is ON. 

Debug mode is when the following occurs...  Chassis is ON and Host is OFF.. 

```
root@frame6u11-bmc:~# obmcutil state 
CurrentBMCState:    xyz.openbmc_project.State.BMC.BMCState.Ready
CurrentPowerState:  xyz.openbmc_project.State.Chassis.PowerState.On
CurrentHostState:   xyz.openbmc_project.State.Host.HostState.Off
root@frame6u11-bmc:~# 
```

**Comment by gurevichmark** [comment]
Do you mean we just want to return something like:
```
[root@stratton01 xcat]# rpower p9euh01 bmcstate
p9euh01: BMC Service
[root@stratton01 xcat]#
```

Or should we always check for BMC state and not allow some `rpower` commands to go through when BMC is in service state ?

**Comment by whowutwut** [comment]
@gurevichmark No, the rpower command should detect when it's in this "debug" state when query `state|stat`

Currently, when it's in that state, `rpower <node> state` returns the server is "On".  But the compute side is not booted.  So this is misleading because we will think that all your compute nodes are Running, but when you ssh to the compute node, it won't work and console will not have anything...  because the host is not booted, but we report "On" as the state.

**Comment by zet809** [comment]
hi, @whowutwut and @gurevichmark , we have `rpower bmcstate` to check the status of BMC like this, so can we close this issue?
```
[root@briggs01 ~]# rpower p9up bmcstate
mid05tor12cn18: Error: BMC did not respond within 10 seconds, retry the command.
mid05tor12cn13: BMC Ready
mid05tor12cn16: BMC Ready
mid05tor12cn11: BMC Ready
mid05tor12cn15: BMC Ready
mid05tor12cn02: BMC Ready
mid05tor12cn05: BMC Ready
```

**Comment by whowutwut** [comment]
This was resolved by  #4047 using commit 5b6074c9b490047169729b8aea1816f0a6b0be16

**Comment by whowutwut** [comment]
@zet809 This is actually a state in the BMC when the Host is OFF, but the Chassis is ON.  It's a debug state and represented by `on (Chassis)`
